src/stories/modal.stories.tsx

```diff
@@ -70,6 +71,63 @@ function validation(quantity: string | number): string {
     return null;
 }
 
+stories.add('Multi popup', withState({ step: 1 })
+    (withInfo('Multi step popup modal')
+        (({ store }) =>
+            <Popup
+                headerText={'Multi step popup ' + store.state.step} 
+                bodyText={getBodyText(store.state.step)}
+                primaryBtnText={getPrimaryBtnText(store.state.step)}
+                primaryBtnHandler={() => store.set({ step: validateStep(store.state.step + 1, 3) })}
+                primaryBtnStyle="primary"
+                secondaryBtnText={getSecondaryBtnText(store.state.step)}
+                secondaryBtnHandler={() => store.set({ step: validateStep(store.state.step - 1, 3) })}
+                secondaryBtnStyle="quiet"
+                step={store.state.step}
+                totalSteps={3} />
+        )
+    )
+);
+
+function getBodyText(step: number): string {
+    switch (step) {
+        case 1:
+            return 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
+        case 2:
+            return 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
+        case 3:
+            return 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec efficitur lectus lorem, id consectetur tellus sollicitudin eu. Curabitur nec metus nibh. Sed sagittis mi nisi, sollicitudin viverra magna malesuada nec. Morbi porttitor, nunc nec mollis feugiat, ex felis commodo sapien, vitae porta sapien magna sit amet nibh. Fusce quis porttitor massa. Pellentesque ornare risus at elementum tempor. Sed convallis odio tellus. Aenean tincidunt purus ut nisl ultrices, sit amet auctor odio porta. Mauris euismod ligula luctus semper dictum. Nulla orci leo, facilisis ac dapibus vitae, consequat ut sem.'
+    }
+}
+
+function getPrimaryBtnText(step: number): string {
```

Comment:
These 3 functions could be a default implementation, since I think they would otherwise be repeated. Of course they would have to accept any number of steps and be translated..

Reply:
I think the current solution gives the most flexibility